### ISSUE TITLE
constexpr destructors not recognized

### ISSUE THREAD

**Issue by arludwig** [microsoft/vscode-cpptools#7965]
Bug type: Language Service

**Describe the bug**
- OS and Version: Windows 10 Pro 20H2
- VS Code Version: 1.59.0
- C/C++ Extension Version: 1.5.1
- Other extensions you installed (and if the issue persists after disabling them): None
- If using SSH remote, specify OS of remote machine: -
- A clear and concise description of what the bug is, including information about the workspace (i.e. is the workspace a single project or multiple projects, size of the project, etc):

main.cpp:
```
struct Foo
{
    constexpr ~Foo() { }  // when you comment this out, everything is fine
};

constexpr auto cfoo = Foo();  // "Foo::~Foo" not defined
auto foo = Foo();            // this is ok though

int main()
{
    return 0;
}
```

When I put a custom `constexpr` destructor in a class, as C++20 now allows us to do, Intellisense says it were not defined.

To reproduce, use above code.

GCC 10.2 (both from recent Mingw and from the Gnu Arm Embedded Toolchain) compiles this code without issues.

Expected behavior is that Intellisense would not issue an error here.

settings.json:
```
{
    "C_Cpp.default.configurationProvider": "ms-vscode.cmake-tools"
}
```

No other files present, except for an CMakeLists.txt
```
cmake_minimum_required(VERSION 3.13.1)
project(FooExample VERSION 0.1.0)

set(CMAKE_CXX_STANDARD 20)
set(CMAKE_CXX_EXTENSIONS OFF)

add_executable(app)

target_sources(app PRIVATE
    "main.cpp"
)
```

I used cmake 3.17.5 to configure and Ninja to build.

Screenshot: (my Intellisense is in german though)
![FooExample](https://user-images.githubusercontent.com/60502779/129226211-a3f8c47c-a6ed-449e-9f29-f1fab192be3f.png)

**Comment by michelleangela** [comment]
@Colengms is this a duplicate of https://github.com/microsoft/vscode-cpptools/issues/4736?

**Comment by Colengms** [comment]
@michelleangela I don't think so.  #4736 is a special case where the the compiler decides to consider an intrinsic function a constant expression at compile time.

This repro's in VS as well, so looks like an IntelliSense bug.

**Comment by michelleangela** [comment]
@Colengms thanks, I created a Visual Studio feedback ticket for this issue because this feature is shared between the C++ extension and Visual Studio's IntelliSense for C++.
https://developercommunity.visualstudio.com/t/constexpr-destructors-not-recognized/1502282

@arludwig Please refer to the Visual Studio ticket for updates on the issue as well as to upvote the issue to increase its priority.

**Comment by michelleangela** [comment]
@arludwig 

**Update:** I closed the Visual Studio ticket https://developercommunity.visualstudio.com/t/constexpr-destructors-not-recognized/1502282. We now have a different internal way to keep track of bugs that are shared between the C++ extension and Visual Studio.

I apologize for any confusion.
